plutus-contract/src/Plutus/Contract/Trace/RequestHandler.hs

```diff
@@ -150,10 +151,10 @@ handleTimeNotifications =
     RequestHandler $ \targetTime_ ->
         surroundDebug @Text "handleTimeNotifications" $ do
             currentSlot <- Wallet.Effects.getClientSlot
-            let targetSlot_ = TimeSlot.posixTimeToSlot targetTime_
+            let targetSlot_ = TimeSlot.posixTimeToSlot def targetTime_
             logDebug $ SlotNoticationTargetVsCurrent targetSlot_ currentSlot
             guard (currentSlot >= targetSlot_)
-            pure $ TimeSlot.slotToPOSIXTime currentSlot
+            pure $ TimeSlot.slotToBeginPOSIXTime def currentSlot
```

Comment:
Not sure if the correct call is `slotToBeginPOSIXTime` or `slotToEndPOSIXTime` (or even `slotToPOSIXTimeRange`)

This function is used by `awaitTime`. So, if currentSlot is 0 and slot length is 2, then `awaitTime 1` will return time of 0. Doesn't make too much sense, and the same issue arises with `slotToEndPOSIXTime`.

Reply:
Well, I guess the idea is that we return the "current" time. Or perhaps more generally: the latest time which we can guarantee has passed. So that sounds like the end of the slot to me.

Reply:
Makes sense. Thanks!